src/types/index.ts

```diff
@@ -3,7 +3,7 @@ import { TextStyle, TouchableOpacityProps, ViewStyle } from 'react-native';
 
 export type ReactChildren = React.ReactNode;
 
-export type ToastType = string;
+export type ToastType = 'success' | 'error' | 'info' | string;
```

Comment:
Hi @ShivamJoker, unfortunately this does not help: adding `| string` at the end widens the type of `ToastType` to `string`.

Reply:
@calintamas there should be some way to have all the types ?

Reply:
I suppose it should dynamically extract all the possible types from the `config` prop

Reply:
No it doesn't give any auto completion

Reply:
Indeed, it's implemented as a simple `string` at this point. I was just suggesting how I think it can be improved

Reply:
You know what I mean?

`ToastType` to be dynamically computed depending on the `config` prop.
For example:

```js
const toastConfig = {
  foo: () => {},
  bar: () => {},
  baz: () => {}
}

<Toast config={toastConfig} />
```
Would result in:
```js
export type ToastType = 'foo' | 'bar' | 'baz';
```

Reply:
Oh I see! Here is the issue which I was going through to see if we can do anything about it https://github.com/microsoft/TypeScript/issues/29729

I am not sure how will we dynamically generate it, but in redux toolkit they are using some stuffs to do dynamically generate types.

Reply:
A solution would be to take a generic.

```tsx
type ToastTypes = 'success' | 'error' | 'notification';

const toastConfig: ToastConfig<ToastTypes> = {
success: () => {},
error: () => {},
notification: () => {}
}

<Toast<ToastTypes> config={toastConfig} />
```

ToastConfig would then be

```ts
type ToastConfig<ToastTypes extends *something to ensure it is a string union*> = {
    [key: ToastTypes]: (params: ToastConfigParams<any>) => React.ReactNode;
};
```

But this doesn't really solve the Toast.show() autocomplete.

An other way would be to allow users to extends the type of the library as I do with react-navigation:

```ts
// Globally type react-navigation
declare global {
  // eslint-disable-next-line @typescript-eslint/no-namespace
  namespace ReactNavigation {
    interface RootParamList extends RoutesParams {}
  }
}
```

On this example I set the global `RootParamList` to my `RoutesParams` so my `navigation.navigate()` have proper autocomplete.

An other way to do this without global types editing is to hint the user to make a wrapper. They create the following function with the generic I proposed above:

```ts
export function toastShow(arg) {
  return Toast.show<ToastTypes>(arg);
}
```

This wrapper only provide the type, as we could do with useNavigation without editing the global types:

```ts
export function useNavigation() {
  return useNavigation<MyNavigationTypes>();
}
```